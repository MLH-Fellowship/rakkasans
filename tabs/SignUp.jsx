// components/signup.js
import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  StatusBar,
  View,
  TextInput,
  Button,
  Image,
  Alert,
  ActivityIndicator,
  TouchableOpacity,
} from "react-native";
import Firebase from "../constants/FireBaseDb";
import Colors from "../constants/Colors";

export default class Signup extends Component {
  constructor() {
    super();
    this.state = {
      displayName: "",
      email: "",
      password: "",
      isLoading: false,
    };
  }

  updateInputVal = (val, prop) => {
    const state = this.state;
    state[prop] = val;
    this.setState(state);
  };

  registerUser = () => {
    if (this.state.email === "" && this.state.password === "") {
      Alert.alert("Enter details to sign up!");
    } else {
      this.setState({
        isLoading: true,
      });
      Firebase.auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then((res) => {
          res.user.updateProfile({
            displayName: this.state.displayName,
          });
          console.log("User registered successfully!");
          this.setState({
            isLoading: false,
            displayName: "",
            email: "",
            password: "",
          });
          this.props.navigation.navigate("Tab Bar");
        })
        .catch((error) => this.setState({ errorMessage: error.message }));
    }
  };

  render() {
    if (this.state.isLoading) {
      return (
        <View style={styles.preloader}>
          <ActivityIndicator size="large" color={Colors.primary} />
        </View>
      );
    }
    return (
      <View style={styles.container}>
        <StatusBar barStyle={"dark-content"} />
        <View style={{ alignItems: "center", marginBottom: 20 }}>
          <Image
            source={require("../assets/images/Torri.png")}
            style={{
              width: 200,
              height: 200,
            }}
          />
          <Text style={styles.logoText}>RAKKASANS</Text>
        </View>
        <View style={styles.textInputContainer}>
          <TextInput
            style={styles.textInput}
            placeholder="Name"
            value={this.state.displayName}
            autoCorrect={false}
            autoCapitalize="none"
            onChangeText={(val) => this.updateInputVal(val, "displayName")}
          />
          <TextInput
            style={styles.textInput}
            placeholder="Email"
            value={this.state.email}
            autoCorrect={false}
            autoCapitalize="none"
            onChangeText={(val) => this.updateInputVal(val, "email")}
          />
          <TextInput
            style={styles.textInput}
            placeholder="Password"
            value={this.state.password}
            onChangeText={(val) => this.updateInputVal(val, "password")}
            maxLength={15}
            autoCorrect={false}
            autoCapitalize="none"
            secureTextEntry={true}
          />
        </View>
        <TouchableOpacity
          style={styles.button}
          onPress={() => this.registerUser()}
        >
          <Text style={styles.buttonText}>SIGN UP</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => this.props.navigation.navigate("Login")}
        >
          <Text style={styles.buttonText}>RETURN TO LOGIN</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    backgroundColor: "#fff",
  },
  button: {
    backgroundColor: Colors.primary,
    height: 60,
    marginHorizontal: 30,
    borderRadius: 35,
    alignItems: "center",
    justifyContent: "center",
    marginVertical: 5,
    shadowOffset: { width: 2, height: 2 },
    shadowColor: "black",
    shadowOpacity: 0.2,
    marginTop: 10,
  },
  buttonText: {
    color: "white",
    fontSize: 20,
    fontWeight: "bold",
  },
  inputStyle: {
    width: "100%",
    marginBottom: 15,
    paddingBottom: 15,
    alignSelf: "center",
    borderColor: "#ccc",
    borderBottomWidth: 1,
  },
  textInputContainer: {
    marginBottom: 40,
  },
  preloader: {
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    position: "absolute",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#fff",
  },
  textInput: {
    height: 50,
    borderRadius: 25,
    borderWidth: 0.5,
    marginHorizontal: 20,
    paddingLeft: 10,
    marginVertical: 8,
    borderColor: "rgba(0,0,0,0.2)",
  },
  logoText: {
    fontSize: 30,
    paddingLeft: 25,
  },
});
