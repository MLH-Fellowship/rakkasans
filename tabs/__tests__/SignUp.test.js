import React from 'react';
import renderer from 'react-test-renderer';

import SignUp from '../SignUp';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<SignUp />).toJSON();
  expect(tree).toMatchSnapshot();
});
