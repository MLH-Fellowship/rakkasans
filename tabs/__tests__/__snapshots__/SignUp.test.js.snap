// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "fira-sans",
          "fontSize": 30,
          "paddingLeft": 25,
        }
      }
    >
      RAKKASANS
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 40,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Name"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 25,
          "borderWidth": 0.5,
          "height": 50,
          "marginHorizontal": 20,
          "marginVertical": 8,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 25,
          "borderWidth": 0.5,
          "height": 50,
          "marginHorizontal": 20,
          "marginVertical": 8,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      maxLength={15}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 25,
          "borderWidth": 0.5,
          "height": 50,
          "marginHorizontal": 20,
          "marginVertical": 8,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#020a41",
        "borderRadius": 35,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginTop": 10,
        "marginVertical": 5,
        "opacity": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "fira-sans",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      SIGN UP
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#020a41",
        "borderRadius": 35,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginTop": 10,
        "marginVertical": 5,
        "opacity": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "fira-sans",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      RETURN TO LOGIN
    </Text>
  </View>
</View>
`;
